lib/liquid/tags/for.rb

```diff
@@ -148,7 +148,7 @@ def render_segment(context, segment)
 
       result = ''
 
-      context.stack do
+      context.stack('forloop', @variable_name) do
```

Comment:
Shouldn't `'forloop'` be frozen here..?
```suggestion
      context.stack('forloop'.freeze, @variable_name) do
```